### DISCUSSION TITLE
I want to query public variable names on-chain

### DISCUSSION THREAD

**Discussion by Olanetsoft** [wevm/wagmi#2495] [Q&A]
`wagmi` uses:

const { data: useContractReadData } = useContractRead({
    address: AVALANCHE_CONTRACT_ADDRESS,
    abi: SendMessageContract.abi,
    functionName: "getValue",
    watch: true,
  });

To read function, but I want to read a public variable in the contract, how can I achieve that?

**Comment by nandochen** [comment]
You can simply put the public variable name into the functionName: 

`
  const { 
data,
   } = useReadContract({
    ...,
    functionName: 'nameOfTheVairable',
    args: [],
  })`

Also, you can pass index, key or whatever the map/array variable uses as index to retrieve item in it: 

`
  const { 
data,
   } = useReadContract({
    ...,
    functionName: 'nameOfTheArrayVairable',
    args: [1],
  })`
